internal/planner/type_join.go

```diff
@@ -377,8 +382,12 @@ func (p *Planner) newInvertableTypeJoin(
 }
 
 type joinSide struct {
-	plan               planNode
-	relFieldDef        client.FieldDefinition
+	plan planNode
+	// The field definition of the relation-object field on this side of the relation.
+	//
+	// This will always have a value on the primary side, but it may not have a value on
+	// the secondary side, as the secondary half of the relation is optional.
```

Comment:
thought: I look forward to the secondary side being handle automatically with the proposed handling of relationship from the primary side.